code/datums/gamemode/role/grue.dm

```diff
@@ -0,0 +1,31 @@
+/datum/role/grue
+	name = GRUE
+	id = GRUE
+	special_role = GRUE
+	required_pref = ROLE_GRUE
+//	wikiroute = ROLE_GRUE
+	logo_state = "grue-logo"
+	default_admin_voice = "Your Grue Instincts"
+	admin_voice_style = "grue"
+	var/eatencount=0
+	var/offspring_born=0
+
+/datum/role/grue/Greet()
+	to_chat(antag.current, "<span class='danger'>You are a grue.</span>")
+	to_chat(antag.current, "<span class='danger'>Darkness is your ally; bright light is harmful to your kind. You hunger... specifically for sentient beings, but you are still young and cannot eat until you are fully mature.</span>")
+	to_chat(antag.current, "<span class='danger'>Bask in shadows to prepare to moult. The more sentient beings you eat, the more powerful you will become.</span>")
+
+/datum/role/grue/ForgeObjectives(var/hatched) //Check if they hatched from an egg or spawned in
+	if(hatched) //Assign it grue_basic objectives if its a hatched grue
+		AppendObjective(/datum/objective/grue/grue_basic)
+	else
+		AppendObjective(/datum/objective/grue/eat_sentients)
+
+/datum/role/grue/GetScoreboard()
+	. = ..()
+	if(istype(antag.current,/mob/living/simple_animal/hostile/grue))
+		. += "<BR>The grue ate [eatencount] sentient beings."
+//		if(config.grue_egglaying)
```

Comment:
It was commented out because I wasn't sure how to actually update these role-defined variables in-game as the grue eats or spawns offspring. Am I correct in thinking it's something like this:
`M.mind.role.eatencount++`
?

Reply:
No. I would be something like this:
```dm
var/datum/role/grue/G = M.mind.GetRole(ROLE_GRUE)
G.eatencount++
```